### PR TITLE
python 3.11 support

### PR DESCRIPTION
**Pull request by ogrodnek**
*Issue #, if available:* #2053

*Description of changes:*

Python 3.11 support
https://aws.amazon.com/blogs/compute/python-3-11-runtime-now-available-in-aws-lambda/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: chalice/deploy/packager.py [hunk 1]
```diff
@@ -498,7 +499,8 @@ class DependencyBuilder(object):
         'cp36m': (2, 17),
         'cp37m': (2, 17),
         'cp38': (2, 26),
-        'cp310': (2, 26)
+        'cp310': (2, 26),
+        'cp311': (2, 26),
```
Comment (ogrodnek) [inline chalice/deploy/packager.py:503]:
```
$ docker run -it --entrypoint "ldd" public.ecr.aws/lambda/python:3.11 --version
ldd (GNU libc) 2.26
```

File: chalice/config.py [hunk 2]
```diff
@@ -151,16 +151,13 @@ def lambda_python_version(self) -> str:
...
         elif (major, minor) <= (3, 6):
             return 'python3.6'
-        elif (major, minor) <= (3, 7):
-            return 'python3.7'
-        elif (major, minor) <= (3, 8):
-            return 'python3.8'
-        elif (major, minor) <= (3, 9):
-            return 'python3.9'
-        return 'python3.10'
+        elif (major, minor) <= (3, 10):
```
Comment (ogrodnek) [inline chalice/config.py:158]: Originally I just added an additional branch, but pylint did not like having this many return statements.

File: docs/source/topics/middleware.rst [hunk 3]
```diff
@@ -271,15 +271,15 @@ Integrating with AWS Lambda Powertools
 --------------------------------------
 
 `AWS Lambda Powertools
-<https://awslabs.github.io/aws-lambda-powertools-python/latest/>`__ is a suite of
+<https://docs.powertools.aws.dev/lambda/python/latest/>`__ is a suite of
```
Comment (ogrodnek) [inline docs/source/topics/middleware.rst:274]: the prcheck link checker complained (failed) about these powertools links.  The redirect seemed ok to me in my browser, but, updated them.

File: tests/integration/test_package.py [hunk 4]
```diff
@@ -45,7 +45,7 @@
         ],
     },
     'numpy': {
-        'version': '1.21.6',
+        'version': '1.23.3',
```
Comment (ogrodnek) [inline tests/integration/test_package.py:48]: had a similar issue to last time where I had to bump these slightly to get a version that included 3.11.

File: chalice/config.py [hunk 5]
```diff
@@ -151,16 +151,13 @@ def lambda_python_version(self) -> str:
         if major == 2:
             return 'python2.7'
         # Python 3 for backwards compatibility needs to select python3.6
-        # for python versions 3.0-3.6. 3.7 and higher will use python3.7.
+        # for python versions 3.0-3.6. 3.7-3.10 will use their version.
+        # 3.11 and higher will use 3.11
         elif (major, minor) <= (3, 6):
             return 'python3.6'
```
Comment (ogrodnek) [inline chalice/config.py:157]: @jamesls I was noticing python 3.6 appears no longer supported: https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

Wasn't sure about changing this to force 3.7 as minimum?  Maybe removing 3.6 would be better as a separate PR if it's wanted?...

**Comment by pietrobolcato** [comment]
This should be merged, is really important that chalice doesn't lag behind so much

**Comment by devinmatte** [comment]
@jamesls Do you have any time soon to take a look at this?

**Comment by semlak** [comment]
Agreed. It is important to merge this or provide feedback.

**Comment by semlak** [comment]
> This should be merged, is really important that chalice doesn't lag behind so much


Agreed. It is important to merge this or provide feedback.

**Comment by ogrodnek** [comment]
Hi @jamesls I just wanted to check in to see if there's any changes I can make to make this PR easier to merge or if you have any feedback, I'm happy to make it.  I know you're busy, so, no worries if you're also not ready to take a look at this.  Thanks, and thanks again for your work on chalice!

**Comment by Steve-Groner** [comment]
@jamesls Any word on merging this so Python 3.11 is supported?

**Comment by venkatajax** [comment]
Can someone pls comment if this PR is merged?

**Comment by rodolfolottin** [comment]
Looking forward for this! 🙌

**Comment by MariaMokbel** [comment]
Can't wait to have this PR merged 🙏🏼

**Comment by hbock-42** [comment]
Seems very promising, huge update, I understand it takes time to review

**Comment by jamesls** [comment]
Taking a look now.

**Comment by jamesls** [comment]
Looks great, thanks for the pull request!

**Comment by ogrodnek** [comment]
Thanks @jamesls , I really appreciate your work on chalice!  It's been really great to use!
